public/components/integrations/components/setup_integration.tsx

```diff
@@ -318,75 +220,69 @@ export function SetupBottomBar({
               window.location.hash = hash;
             }}
           >
-            Cancel
-          </Eui.EuiButton>
-        </Eui.EuiFlexItem>
-        <Eui.EuiFlexItem>
-          <Eui.EuiSpacer />
-        </Eui.EuiFlexItem>
-        {step > 0 ? (
-          <Eui.EuiFlexItem grow={false}>
-            <Eui.EuiButton iconType={'returnKey'} onClick={() => setStep(step - 1)}>
-              Back
-            </Eui.EuiButton>
-          </Eui.EuiFlexItem>
-        ) : null}
-        <Eui.EuiFlexItem grow={false}>
-          <Eui.EuiButton
+            Discard
+          </EuiButton>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButton
             fill
-            iconType={'check'}
-            onClick={() => {
-              if (step < STEPS.length - 1) {
-                setStep(step + 1);
-              } else {
-                console.log(config);
-              }
+            iconType="arrowRight"
+            iconSide="right"
+            isLoading={loading}
+            onClick={async () => {
+              setLoading(true);
+              const template = await findTemplate(integration.name);
+              await addIntegrationRequest(
+                false,
+                integration.name,
+                config.displayName,
+                template,
+                setToast,
+                config.displayName,
+                config.connectionDataSource
+              );
+              setLoading(false);
```

Comment:
do you want to move this to a function?

Reply:
Can you elaborate? Do you mean move this to a separate wrapper function that only gets used here?

Reply:
Yes, move onclick part to a separate function: `onClick={handleAddIntegration}` . It is just 4 statements but linter makes it look like 12 lines. Just a suggestion.